src/wallet/wallet2.h

```diff
@@ -726,6 +732,29 @@ namespace tools
     void scan_output(const cryptonote::account_keys &keys, const cryptonote::transaction &tx, const crypto::public_key &tx_pub_key, size_t i, tx_scan_info_t &tx_scan_info, int &num_vouts_received, uint64_t &tx_money_got_in_outs, std::vector<size_t> &outs);
     void trim_hashchain();
 
+    /*
+     * "extended attributes" are a mechanism to store an arbitrary number of string values
+     * on the level of the wallet as a whole, identified by keys. Their introduction,
+     * technically the unordered map m_extended_attributes stored as part of a wallet file,
+     * led to a new wallet file version, but now new singular pieces of info may be added
+     * without the need for a new version.
+     *
+     * The first and so far only value stored as such an extended attribute is the wallet note.
+     * It's stored under the internal standard key EA_WALLET_NOTE (see method set_wallet_note).
+     *
+     * The mechanism is now private to wallet2. For the future it would be feasible to open
+     * it to all clients however and allow them to use it for storing basically any single
+     * string values in a wallet. To avoid the problem that different clients possibly
+     * overwrite or misunderstand each other's extended attributes, a two-part key is
+     * proposed: <client name>.<value name>
+     *
+     * The term "extended attribute" is borrowed from the domain of file systems where
+     * such attributes allow storing key/value pairs "attached" to files.
```

Comment:
Can you expand on that ? It seems to be an answer to my "why this weird name" question, but it doesn't actually give the actual answer :)

Reply:
Well, file systems on Windows and also some on Linux offer a feature with exactly this name, i.e. *extended attributes*, with name, file size, "normal" attributes like *executable* being "normal" attributes, and anything else that any client might stick to a file and have it stored together with it called extended attributes. E.g. see [Wikipedia](https://en.wikipedia.org/wiki/Extended_file_attributes).

I thought this would be more "meaningful" than something bland and generic like "extra values" or similar. Not sure whether I really succeeded however :)